image/image.go

```diff
@@ -303,6 +303,9 @@ type Info struct {
 	// will be the same for references that point at the same image
 	// (but does not necessarily equal Docker's image ID)
 	ImageID string `json:",omitempty"`
+	// indicator if the usage of the timestamp label is whitelisted
+	// for this image
+	UseTimestampLabels bool `json:",omitempty"`
```

Comment:
This is used as a value in API results (regrettably); changing the meaning of fields means every client, and every intermediary, must have the same idea of how it works. It might be better to just overwrite `CreatedAt` when collating the images -- then clients don't have to know how to interpret. Optionally, a type which serialises the same way could be added to the api package, to right the historical wrong.

What are the consequences of doing it that way?

Reply:
> Optionally, a type which serialises the same way could be added to the api package, to right the historical wrong.

This is the solution I like the most, but I have some questions about what changes are allowed due to versioning / backwards compatibility.

I am happy to make this change in an additional PR after we have discussed this, but as getting this PR in is quite critical due to the impact labels from base images have I implemented the former solution for now.

Reply:
If I'm reading the diff correctly, the created timestamp is overwritten when the registry client analyses the manifest it has fetched; i.e., before an entry is put in memcached. Entries _coming from_ memcached treat the Created field as the timestamp.

This means any entries already in memcached will not consult the labels even if they match a `--registry-use-labels` pattern. Could the images be stored as before, and the timestamp set _when fetched_? That would make existing entries valid as well as new entries.